pkg/k8s/mounterBuilder.go

```diff
@@ -0,0 +1,129 @@
+package k8s
+
+import (
+	"github.com/feedhenry/mcp-standalone/pkg/mobile"
+	"github.com/pkg/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+type MounterBuilder struct {
+	k8s       kubernetes.Interface
+	namespace string
+}
+
+func NewMounterBuilder(namespace string) mobile.MounterBuilder {
+	return &MounterBuilder{namespace: namespace}
+}
+
+func (mb *MounterBuilder) WithK8s(k8s kubernetes.Interface) mobile.MounterBuilder {
+	mb.k8s = k8s
+	return mb
+}
+
+func (mb *MounterBuilder) Build() mobile.VolumeMounterUnmounter {
+	return &MountManager{k8s: mb.k8s, namespace: mb.namespace}
+}
+
+type MountManager struct {
+	k8s       kubernetes.Interface
+	namespace string
+}
+
+// Mount a secret named mount into the service
+func (mm *MountManager) Mount(secret, clientService string) error {
+	if s, err := mm.k8s.CoreV1().Secrets(mm.namespace).Get(secret, meta_v1.GetOptions{}); err != nil || s.Name != secret {
```

Comment:
not sure we need to check the secret names match here

Reply:
the `Get` returns a blank secret if nothing was found, this is checking the returned secret is not a blank secret.

Reply:
are you sure it wont return nil here? it is returning a pointer so I imagine it would return nil

Reply:
if I remove that `|| s.Name != secret`, my test for an error response with a bad secret name fails.